examples/microwave-oven-app/linux/main.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "microwave-oven-device.h"
+#include <AppMain.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+
+static constexpr EndpointId kMicrowwaveOvenEndpointId = 1;
+
+Platform::UniquePtr<MicrowaveOvenDevice> gMicrowaveOvenDevice;
+
+void ApplicationInit()
+{
+    gMicrowaveOvenDevice = Platform::MakeUnique<MicrowaveOvenDevice>(kMicrowwaveOvenEndpointId);
+    gMicrowaveOvenDevice.get()->Init(kMicrowwaveOvenEndpointId);
+}
+
+void ApplicationShutdown() {}
```

Comment:
Any reason this is not effectively shutting down gMicrowaveOvenDevice (if not null) and just leaking it instead?

Reply:
`gMicrowaveOvenDevice` is defined as an unqiue pointer(above line 27), this could be release the memory automatically when the program is shutdown unexpectedly. 

But programs will always run to `ApplicationShutdown()` finally, we should keep it.

Reply:
My point is this: ApplicationShutdown should shut down the things ApplicationInit inits.